### ISSUE TITLE
Kokkos:  std algorithms installed twice

### ISSUE THREAD

**Issue by maartenarnst** [trilinos/Trilinos#11358]
## Bug Report
@trilinos/kokkos

### Description
The headers for the std algorithms are installed twice.  Once in the main trilinos include dir, and then once more in a subdir `std_algorithms`.  The issue might have something to do with how the install is configured (when kokkos is part of trilinos, and in particular with regards to the `std_algorithms` subdir) in `packages/kokkos/algorithms/src/CMakeLists.txt`.

This is a sample of output from the installation:

```
 -- Installing: /opt/Trilinos/GNU-Cuda/include/std_algorithms/Kokkos_BeginEnd.hpp
...
 -- Installing: /opt/Trilinos/GNU-Cuda/include/Kokkos_BeginEnd.hpp
```

### Steps to Reproduce
This is our configure : [CMakePresets.json.txt](https://github.com/trilinos/Trilinos/files/10176279/CMakePresets.json.txt)

**Comment by rppawlo** [comment]
@ndellingwood

**Comment by ndellingwood** [comment]
Thanks for reporting, I opened a Kokkos issue for tracking as well with kokkos/kokkos#5667

**Comment by nmm0** [comment]
A fix for this was [merged[(https://github.com/kokkos/kokkos/pull/5670) into Kokkos develop

**Comment by GrahamBenHarper** [comment]
This was fixed in the 4.0.0 snapshot #11600
